src/bp/core/services/workspace-service.ts

```diff
@@ -59,6 +43,15 @@ export class WorkspaceService {
     return this.ghost.global().upsertFile('/', `workspaces.json`, JSON.stringify(workspaces, undefined, 2))
   }
 
+  async mergeWorkspaceConfig(workspaceId: string, partialData: Partial<Workspace>) {
+    const workspaces = await this.getWorkspaces()
+    if (!workspaces.find(x => x.id === workspaceId)) {
+      throw new NotFoundError(`Workspace doesn't exist`)
+    }
+
+    return this.save(workspaces.map(wks => (wks.id === workspaceId ? { ...wks, ...partialData } : wks)))
```

Comment:
depending on how you want this to behave, this won't merge deep, partial data. Object.assign or using spread operator  is only a single level deep.

Reply:
Really basic for now, it's only to merge top-level properties (name, description and pipeline